pkg/storage/sqlite/sqlite.go

```diff
@@ -1078,6 +1076,30 @@ func (m *SQLite) IsReady(ctx context.Context) (storage.ReadinessStatus, error) {
 	return sqlcommon.IsReady(ctx, m.db)
 }
 
+func HandleSQLError(err error, args ...interface{}) error {
+	if errors.Is(err, sql.ErrNoRows) {
+		return storage.ErrNotFound
+	}
+
+	if errors.Is(err, storage.ErrIteratorDone) {
+		return err
```

Comment:
looks like this is never hit and can be removed :) the iterator is the only struct that returns this err